src/Core/Domain/Carrier/QueryResult/EditableCarrier.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\QueryResult;
+
+/**
+ * Stores carrier data that's needed for editing.
+ */
+class EditableCarrier
+{
+    public function __construct(
+        private int $carrierId,
+        private string $name,
+        private int $grade,
+        private string $trackingUrl,
+        private int $position,
+        private bool $active,
+        /** @var string[] $delay */
+        private array $delay
+    ) {
+    }
+
+    public function getCarrierId(): int
+    {
+        return $this->carrierId;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getGrade(): int
+    {
+        return $this->grade;
+    }
+
+    public function getTrackingUrl(): string
+    {
+        return $this->trackingUrl;
+    }
+
+    public function getPosition(): int
+    {
+        return $this->position;
+    }
+
+    public function isActive(): bool
+    {
+        return $this->active;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getLocalizedDelay(): array
+    {
+        return $this->delay;
+    }
+
+    public function getLogoPath(): ?string
```

Comment:
This logic shouldn't be hard coded in the QueryResult which is supposed to be a simple DTO It's the handler that should handle this logic and pass it as an argument to the DTO

Reply:
Also you're using `_PS_SHIP_IMG_DIR_` so it will return the absolute file path on the server, which is maybe not that useful, the URL will probably be more interesting for both the BO form and the future API based on this query

Basically it's probably preferable to use the `_THEME_SHIP_DIR_` constant instead which should contain the base URL for carrier images (but of course `_PS_SHIP_IMG_DIR_` is still useful to check the file existence)